tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -0,0 +1,526 @@
+package org.unicode.cldr.util.personname;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.TreeSet;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.ibm.icu.util.ULocale;
+
+/**
+ * Rough sketch for now
+ * TODO Make classes/methods private that don't need to be public
+ * TODO Check for invalid parameters
+ */
+
+public class PersonNameFormatter {
+    public enum Field {
+        prefix,
+        given,
+        given2,
+        surname,
+        surname2,
+        suffix;
```

Comment:
I thought standard Java style was that enum members had names in ALL CAPS.  Did that change at some point after I quit writing Java?

Reply:
Maybe that's the answer to the "long" problem. :)